### PR TITLE
Enable nullable reference types for Roslyn.Services.UnitTests.Utilities

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Workspaces/CoreTestUtilities/Formatting/FormattingTestBase.cs [hunk 1]
```diff
@@ -46,7 +48,7 @@ public abstract class FormattingTestBase
 
                 var document = project.AddDocument("Document", SourceText.From(code));
 
-                var syntaxTree = await document.GetSyntaxTreeAsync();
+                var syntaxTree = (await document.GetSyntaxTreeAsync())!;
```
Comment (jasonmalinowski) [inline src/Workspaces/CoreTestUtilities/Formatting/FormattingTestBase.cs]: We've got GetRequiredSyntaxTreeAsync().

File: src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs [hunk 2]
```diff
@@ -90,8 +92,7 @@ public override void After(MethodInfo methodUnderTest)
             var exportProvider = ExportProviderCache.ExportProviderForCleanup;
             try
             {
-                var listenerProvider = exportProvider?.GetExportedValues<IAsynchronousOperationListenerProvider>().SingleOrDefault();
-                if (listenerProvider != null)
+                if (exportProvider?.GetExportedValues<IAsynchronousOperationListenerProvider>().SingleOrDefault() is { } listenerProvider)
```
Comment (jasonmalinowski) [inline src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs]: I still hate this syntax. :smile:

File: src/Workspaces/CoreTestUtilities/Formatting/FormattingTestBase.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/Formatting/FormattingTestBase.cs]: Fixed this and `GetProject`&rarr;`GetRequiredProject`

File: src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs (hunk 2 above)
Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/MEF/UseExportProviderAttribute.cs]: It was needed here for nullable walker.

**Pull request merged**